easybuild/easyblocks/generic/juliapackage.py

```diff
@@ -96,11 +96,14 @@ def install_step(self):
         """Install Julia package with Pkg"""
 
         # prepend installation directory to Julia DEPOT_PATH
+        # extensions in a bundle can share their DEPOT_PATH
         # see https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_DEPOT_PATH
-        depot_path = os.getenv('JULIA_DEPOT_PATH')
-        if depot_path is not None:
-            depot_path = ':'.join([self.installdir, depot_path])
-        env.setvar('JULIA_DEPOT_PATH', depot_path)
+        depot_path = os.getenv('JULIA_DEPOT_PATH', '')
+        if depot_path:
+            depot_path = depot_path.split(os.pathsep)
+        if self.installdir not in depot_path:
+            depot_path = os.pathsep.join([depot for depot in (self.installdir, depot_path) if depot])
+            env.setvar('JULIA_DEPOT_PATH', depot_path)
```

Comment:
this will fail if depot_path exits, because then it's converted from a string to a list.
```suggestion
        depot_path = os.getenv('JULIA_DEPOT_PATH', [])
        if depot_path:
            depot_path = depot_path.split(os.pathsep)
        if self.installdir not in depot_path:
            depot_path = os.pathsep.join([depot for depot in [self.installdir] + depot_path if depot])
            env.setvar('JULIA_DEPOT_PATH', depot_path)
```